src/test/scala/io/iohk/ethereum/vm/OpCodeFunSpec.scala

```diff
@@ -251,28 +251,33 @@ class OpCodeFunSpec extends FunSuite with OpCodeTesting with Matchers with Prope
     }
   }
 
-  ignore("BLOCKHASH") {
-    // to be implemented
-  }
-
-  ignore("COINBASE") {
-    // to be implemented
-  }
-
-  ignore("TIMESTAMP") {
-    // to be implemented
-  }
+  test(BLOCKHASH) { op =>
+    val stateGen: Gen[PS] = for {
+      stateIn <- getProgramStateGen(
+        stackGen = getStackGen(maxWord = DataWord(256)),
+        codeGen = getByteStringGen(0, 256),
+        blockNumberGen = getBigIntGen(0, 256)
```

Comment:
With `stackGen` and `blockNumberGen` we will never have a case of the block being older than 256. Maybe use 512 as max for both (although that will result in a requested number out of bounds more often than not).

`codeGen` is not needed.